src/main/java/org/littil/api/guestTeacher/service/GuestTeacherService.java

```diff
@@ -85,9 +94,13 @@ public void deleteTeacher(@NonNull final UUID id, UUID userId) {
     }
 
     private GuestTeacher update(@Valid GuestTeacher guestTeacher) {
+        UUID userId = tokenHelper.getCurrentUserId();
         GuestTeacherEntity entity = repository.findByIdOptional(guestTeacher.getId())
                 .orElseThrow(() -> new NotFoundException("No Teacher found for Id"));
 
+        if (entity.getUser() != null && !entity.getUser().getId().equals(userId)) {
```

Comment:
You already know the user you want to update and your current userId in the resource. Would this be a better place block this request?

Reply:
In the resource we receive a GuestTeacherPostResource which doesn't include the user. Only in the service we fetch the GuestTeacherEntity which includes the user.